sdk/core/core/inc/az_http.h

```diff
@@ -106,11 +106,10 @@ typedef enum
 } az_http_status_code;
 
 /**
- * @brief An az_http_policy_retry_options instance allows you to customize the retry policy
- * used by an XxxClient type whenever it performs an I/O operation. Applications should
- * acquire an initialized instance of this struct by calling az_http_policy_retry_options_default
```

Comment:
Btw, `_az_http_policy_retry_options_default` is an internal API, so how is the caller supposed to create and modify the retry_options? We can't refer to an internal method within the docs.

https://github.com/Azure/azure-sdk-for-c/blob/a941c50ffa218d3a70db99fa0347cacca031e9cb/sdk/core/core/src/az_http_policy_retry.c#L30-L39

cc @JeffreyRichter, @antkmsft

Reply:
Customers do not call this function directly. Instead, when using a specific client library (like Storage), the customer call the client's *_options_default function and it internally calls this function. See https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/storage/blobs/src/az_storage_blobs_blob_client.c#L32

The client library's function needs to document its function. But, the retry structure is publicly exposed/documented from the Core library: https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/core/core/inc/az_http.h#L109.

Reply:
In addition to that, customers aren't supposed to create pipelines, and it is the pipeline where you pass these retry options.

Reply:
So, are the doc changes fine then? Or do you want me to reword it?

Reply:
Docs for this function? You can change it to say that "Client libraries should acquire..."